File: python/read_world_data.py
#-----------------------------------------------------------------------------------
# RBF Python3 library example.
#
# reads a RBF file and print-out record as CSV data
#
#-----------------------------------------------------------------------------------

# standard imports
import sys
from datetime import datetime, date, time

# import our lib
import rbf

# test arguments
if len(sys.argv) == 1:
    print("Usage: %s <xml layout> <rbf file>\n" % sys.argv[0])
    sys.exit()

# save arguments
xml_layout = sys.argv[1]
rb_file = sys.argv[2]

# create new mapper for layout object. It is necessary to define this map to read a RBF file
mapper = lambda x: x[0:4]

# create new layout
layout = rbf.Layout(xml_layout)

# create new reader
reader = rbf.Reader(rb_file, layout, mapper)

# now read each record
for rec in reader:
    print(";".join(rec.array_of('value')))

File: ruby/read_world_data.rb
require "rbf"

# first step: create a Layout object which reads the XML definition file and creates Record and Field objects
# to play with. A Layout object is a hash of record, with the key being the record name, and the value the
# Record object
layout = Layout.new("../data/world/world_data.xml")

# now create a Reader object: it just read every line of the input record-based file and maps it to a Record object from
# the Layout hash. If the next line is of the same Record type, it will be overwritten.
# The 3rd argument is a function or a lambda mapping the line to a record type.
reader = Reader.new("../data/world/world_data.txt", layout, lambda {|x| x[0..3]})

# now, read each record. Next line for example build a CSV file from the rbf input file **world_data.txt**
reader.each {|rec|
  puts rec.array_of(:@value).join(";") 
}